File: src/main/java/com/api/viewdata/service/ViewDataService.java
package com.api.viewdata.service;

import org.springframework.stereotype.Service;
import com.api.viewdata.dto.ViewDataResponse;
import com.api.viewdata.model.Transaction;
import com.api.viewdata.model.Status;
import com.api.viewdata.repo.TransactionRepo;
import com.api.viewdata.repo.StatusRepo;

import java.time.format.DateTimeFormatter;
import java.util.ArrayList;

@Service
public class ViewDataService {
    private final TransactionRepo txRepo;
    private final StatusRepo statusRepo;
    private static final DateTimeFormatter FMT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    public ViewDataService(TransactionRepo txRepo, StatusRepo statusRepo) {
        this.txRepo = txRepo;
        this.statusRepo = statusRepo;
    }

    public ViewDataResponse buildResponse() {
        var res = new ViewDataResponse();

        res.data = new ArrayList<>();
        for (Transaction t : txRepo.findAll()) {
            var row = new ViewDataResponse.Row();
            row.id = t.getId();
            row.productID = t.getProductId();          // exact field name
            row.productName = t.getProductName();
            row.amount = t.getAmountText();            // string
            row.customerName = t.getCustomerName();
            row.status = t.getStatus().getId();
            row.transactionDate = t.getTransactionDate().format(FMT);
            row.createBy = t.getCreateBy();
            row.createOn = t.getCreateOn().format(FMT);
            res.data.add(row);
        }

        res.status = new ArrayList<>();
        for (Status s : statusRepo.findAll()) {
            var si = new ViewDataResponse.StatusItem();
            si.id = s.getId();
            si.name = s.getName();
            res.status.add(si);
        }
        return res;
    }
}

File: internal/service/viewdata_service.go
package service

import (
	"fmt"
	"time"

	"internship-test-api/internal/handler"
	"internship-test-api/internal/model"
	"internship-test-api/internal/repository"
)

type ViewDataService struct {
	statusRepo      *repository.StatusRepository
	transactionRepo *repository.TransactionRepository
}

func NewViewDataService(statusRepo *repository.StatusRepository, transactionRepo *repository.TransactionRepository) *ViewDataService {
	return &ViewDataService{
		statusRepo:      statusRepo,
		transactionRepo: transactionRepo,
	}
}

func (s *ViewDataService) GetViewData() (*handler.ViewDataResponse, error) {
	txs, err := s.transactionRepo.FindAll()
	if err != nil {
		return nil, err
	}

	statuses, err := s.statusRepo.FindAll()
	if err != nil {
		return nil, err
	}

	response := &handler.ViewDataResponse{
		Data:   make([]handler.Row, 0, len(txs)),
		Status: make([]handler.StatusItem, 0, len(statuses)),
	}

	for _, tx := range txs {
		row := handler.Row{
			ID:              tx.ID,
			ProductID:       tx.ProductID,
			ProductName:     tx.ProductName,
			Amount:          tx.AmountText,
			CustomerName:    tx.CustomerName,
			Status:          tx.StatusID,
			TransactionDate: tx.TransactionDate.Format("2006-01-02 15:04:05"),
			CreateBy:        tx.CreateBy,
			CreateOn:        tx.CreateOn.Format("2006-01-02 15:04:05"),
		}
		response.Data = append(response.Data, row)
	}

	for _, st := range statuses {
		response.Status = append(response.Status, handler.StatusItem{
			ID:   st.ID,
			Name: st.Name,
		})
	}

	return response, nil
}
